src/Sulu/Bundle/AdminBundle/Resources/js/views/FormOverlayList/FormOverlayList.js

```diff
@@ -0,0 +1,188 @@
+// @flow
+import React, {Fragment} from 'react';
+import {observer} from 'mobx-react';
+import {action, observable, toJS} from 'mobx';
+import type {ElementRef} from 'react';
+import type {ViewProps} from '../../containers/ViewRenderer';
+import Overlay from '../../components/Overlay';
+import {translate} from '../../utils/Translator';
+import Form from '../../containers/Form';
+import ResourceStore from '../../stores/ResourceStore';
+import List from '../List';
+import ResourceFormStore from '../../containers/Form/stores/ResourceFormStore';
+import Snackbar from '../../components/Snackbar';
+import formOverlayListStyles from './formOverlayList.scss';
+
+@observer
+export default class FormOverlayList extends React.Component<ViewProps> {
+    static getDerivedRouteAttributes = List.getDerivedRouteAttributes;
+
+    listRef: ?ElementRef<typeof List>;
+    formRef: ?ElementRef<typeof Form>;
+
+    @observable formStore: ?ResourceFormStore;
+    @observable formErrors = [];
+
+    handleItemAdd = () => {
+        this.createFormOverlay(undefined);
+    };
+
+    handleItemClick = (itemId: string | number) => {
+        this.createFormOverlay(itemId);
+    };
+
+    handleFormOverlayConfirm = () => {
+        if (!this.formRef) {
+            throw new Error('The Form ref has not been set! This should not happen and is likely a bug.');
+        }
+
+        this.formRef.submit();
+    };
+
+    handleFormOverlayClose = () => {
+        this.destroyFormOverlay();
+    };
+
+    handleFormSubmit = () => {
+        if (!this.formStore) {
+            throw new Error('The FormStore has not been initialized! This should not happen and is likely a bug.');
+        }
+
+        this.formStore.save()
+            .then(() => {
+                this.destroyFormOverlay();
+                if (this.listRef) {
+                    this.listRef.listStore.sendRequest();
```

Comment:
Shouldn't we at least introduce a a `listRef.reload` method? Then it would be more similar to `formRef.submit` and less dependent on the `ListStore`.

Reply:
Of course I would like to not have to use a ref here, but I don't know a better solution at the moment....

Reply:
refactored the reload method of the `ListStore` and added a reload method to the `List` view